src/Arcus.Observability.Telemetry.AspNetCore/Extensions/ILoggerHttpDependencyExtensions.cs

```diff
@@ -185,11 +154,22 @@ public static void LogHttpDependency(
             string dependencyId,
             Dictionary<string, object> context = null)
         {
-            Guard.NotNull(logger, nameof(logger), "Requires a logger instance to track telemetry");
-            Guard.NotNull(request, nameof(request), "Requires a HTTP request message to track a HTTP dependency");
-            Guard.NotLessThan(duration, TimeSpan.Zero, nameof(duration), "Requires a positive time duration of the HTTP dependency operation");
-            Guard.NotLessThan(statusCode, 100, nameof(statusCode), "Requires a valid HTTP response status code that's within the range of 100 to 599, inclusive");
-            Guard.NotGreaterThan(statusCode, 599, nameof(statusCode), "Requires a valid HTTP response status code that's within the range of 100 to 599, inclusive");
+            if (logger is null)
+            {
+                throw new ArgumentNullException(nameof(logger), "Requires a logger instance to track telemetry");
+            }
+            if (request is null)
+            {
+                throw new ArgumentNullException(nameof(request), "Requires a HTTP request message to track a HTTP dependency");
+            }
+            if (duration < TimeSpan.Zero)
+            {
+                throw new ArgumentOutOfRangeException(nameof(duration), "Requires a positive time duration of the HTTP dependency operation");
+            }
+            if (statusCode < 100 || statusCode > 599)
```

Comment:
```suggestion
            }

            if (request is null)
            {
                throw new ArgumentNullException(nameof(request), "Requires a HTTP request message to track a HTTP dependency");
            }

            if (duration < TimeSpan.Zero)
            {
                throw new ArgumentOutOfRangeException(nameof(duration), "Requires a positive time duration of the HTTP dependency operation");
            }

            if (statusCode < 100 || statusCode > 599)
```
**requires change:** since we're following/building on top of Microsoft code style, we should add a blank line between `if` statements.